newdle/api.py

```diff
@@ -307,52 +306,53 @@ def create_newdle(title, duration, timezone, timeslots, participants, private, n
         notify=notify,
     )
     db.session.add(newdle)
+    db.session.flush()
+    send_invitation_emails(newdle)
     db.session.commit()
-    notify_newdle_participants(
-        newdle,
-        f'Invitation: {newdle.title}',
-        'invitation_email.txt',
-        'invitation_email.html',
-        lambda p: {
-            'creator': newdle.creator_name,
-            'title': newdle.title,
-            'participant': p.name,
-            'summary_link': url_for('newdle_summary', code=newdle.code, _external=True),
-            'answer_link': url_for(
-                'newdle', code=newdle.code, participant_code=p.code, _external=True
-            ),
-        },
-    )
     return NewdleSchema().jsonify(newdle)
 
 
-@api.route('/newdle/<code>')
-@allow_anonymous
-def get_newdle(code):
-    newdle = Newdle.query.filter_by(code=code).first_or_404(
-        'Specified newdle does not exist'
-    )
-    if newdle.deleted:
-        return DeletedNewdleSchema().jsonify(newdle)
-    return RestrictedNewdleSchema().jsonify(newdle)
-
-
 @api.route('/newdle/<code>', methods=('PATCH',))
-@use_args(UpdateNewdleSchema(), locations=('json',))
+@use_args(NewNewdleSchema(partial=True), locations=('json',))
```

Comment:
When creating a newdle we currently do not allow anonymous participants (no email/auth_uid). But when editing, we do need to allow such participants since they may already be in the newdle - and right now saving the updated participants fails in that case:

![image](https://user-images.githubusercontent.com/179599/97092084-30106c00-1641-11eb-8d77-e6c76133cfe9.png)